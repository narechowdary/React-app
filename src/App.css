.App {
  text-align: center;
}

.container {
  position: relative;
  text-align: center;
  color: black;
  padding: 0 !important;
  margin-bottom: 30px;
}

.top-left {
  position: absolute;
  top: 8px;
  left: 16px;
}

.product-title {
  width: 100%;
  background-color: #dee2e6;
}

.product-price {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  text-align: left;
  padding: 15px;
}


.hight-price {
  text-decoration: line-through;
  background-color: black;
  color: white;
}

.low-price {
  font-weight: bold;
  background-color: darkgray;
  color: white;
}
