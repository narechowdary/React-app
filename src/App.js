import React from 'react';
import './App.css';
import Main from './Components/src/Main'

export default class extends React.Component{
  constructor(props){
    super(props)
    this.state = { }
  }

  render() {
    return(
      <div>
        {/* Added Product Details main Component */}
        <Main />
      </div>
    )
  }
}
