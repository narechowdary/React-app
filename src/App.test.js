import App from './App';
import renderer from 'react-test-renderer'

test('it renders correctly', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});