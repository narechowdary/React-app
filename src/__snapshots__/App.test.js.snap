// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div>
    <div
      className="App"
    >
      <img
        alt="Williams Sonoma"
        class="logo-image"
        src="https://assets.wsimgs.com/wsimgs/rk/images/i/202045/0007/images/common/logo.svg"
      />
    </div>
     
    <br />
    <div
      className="row"
      tabIndex="-1"
    >
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-cotton-napling-2-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Cotton Napling
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="low-price"
                >
                  $ 
                  149
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-velvet-weighted-blanket-silver-topaz-1-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Velvet Weighted Blanket - Silver Topaz
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  279
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  249
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-velvet-weighted-blanket-2-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Velvet Weighted Blanket - Rose Quartz
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  279
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  249
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-velvet-weighted-blanket-emerald-1-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Velvet Weighted Blanket- Emerald
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  279
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  249
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-velvet-weighted-blanket-pink-sapphire-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Velvet Weighted Blanket - Pink Sapphire
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  279
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  249
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202046/0098/bearaby-velvet-weighted-blanket-turquoise-4-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Bearaby Velvet Weighted Blanket - Turquoise
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  279
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  249
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202043/0001/doctors-without-borders-face-coverings-set-of-2-3-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Doctors Without Borders Face Coverings (Set of 2)
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="low-price"
                >
                  $ 
                  24
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202043/0001/doctors-without-borders-thank-you-ornaments-set-of-3-1-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Doctors Without Borders Thank You Ornaments (Set of 3)
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="low-price"
                >
                  $ 
                  12.5
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202043/0001/doctors-without-borders-thank-you-catchall-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Doctors Without Borders Thank You Catchall
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="low-price"
                >
                  $ 
                  19.5
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="container"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://assets.weimgs.com/weimgs/ab/images/wcm/products/202041/0010/festive-stripe-dinnerware-m.jpg)",
              "backgroundRepeat": "no-repeat",
              "height": 363,
              "width": 363,
            }
          }
        >
          <div
            className="product-details"
          >
            <div
              className="product-title"
            >
              Festive Stripe Dinnerware
            </div>
            <div
              className="product-price"
            >
              <div>
                <span
                  className="hight-price"
                >
                  $ 
                  126
                </span>
                <br />
                <span
                  className="low-price"
                >
                  $ 
                  8.5
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;
